presto-main/src/main/java/io/prestosql/sql/planner/QueryPlanner.java

```diff
@@ -797,6 +797,38 @@ public static Expression coerceIfNecessary(Analysis analysis, Expression origina
                 analysis.isTypeOnlyCoercion(original));
     }
 
+    public static NodeAndMappings coerce(PlanNode node, List<Symbol> fields, List<Type> types, SymbolAllocator symbolAllocator, PlanNodeIdAllocator idAllocator)
+    {
+        Assignments.Builder assignments = Assignments.builder();
+
+        ImmutableList.Builder<Symbol> mappings = ImmutableList.builder();
+        for (int i = 0; i < types.size(); i++) {
+            Symbol input = fields.get(i);
+            Type type = types.get(i);
+
+            if (!symbolAllocator.getTypes().get(input).equals(type)) {
+                Symbol coerced = symbolAllocator.newSymbol(input.getName(), type);
+                assignments.put(coerced, new Cast(input.toSymbolReference(), toSqlType(type)));
```

Comment:
should you use: `io.prestosql.sql.analyzer.Analysis#getTypeOnlyCoercions` here?

Reply:
Type-only coercions don't do anything anymore. They are only used to short-circuit the invocation of the cast in the generated bytecode, but since type-only coercions are just identity methods, that doesn't provide any value -- the JIT will inline the method and remove the call.